src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,148 @@
+package duke.util;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import duke.DukeException;
+import duke.command.AddCommand;
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.EmptyCommand;
+import duke.command.FindCommand;
+import duke.command.HelpCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.UnknownCommand;
+import duke.command.UnmarkCommand;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+/**
+ * Class to parse user input.
+ */
+public class Parser {
+    private static final String DEADLINE_INDICATOR = "\\s*/by\\s*";
+    private static final String EVENT_INDICATOR = "\\s*/at\\s*";
+
+    /**
+     * Parses an input string into a {@code Command} object.
+     *
+     * @param input User input string.
+     * @return {@code Command} object.
+     * @throws DukeException Checked exceptions that may occur when parsing user input into {@code Command}.
+     */
+    public static Command parseCommand(String input) throws DukeException {
+        input = input.strip();
+        String inputCommand = input.indexOf(" ") == -1
+                ? input.toLowerCase() : input.toLowerCase().substring(0, input.indexOf(" "));
+        String[] commandAndArguments = input.split(" ", 2);
+        String args = "";
+        if (commandAndArguments.length > 1) {
+            args = commandAndArguments[1].strip();
+        }
+
+        switch (inputCommand) {
+        case "help":
+            return new HelpCommand();
+        case "bye":
+            return new ByeCommand();
+        case "list":
+            return new ListCommand();
+        case "mark":
+            verifyArgumentsIsNotEmpty(args);
+            return new MarkCommand(parseTaskIndex(args));
+        case "unmark":
+            verifyArgumentsIsNotEmpty(args);
+            return new UnmarkCommand(parseTaskIndex(args));
+        case "todo":
+            verifyArgumentsIsNotEmpty(args);
+            return new AddCommand(parseTodoArguments(args));
+        case "event":
+            verifyArgumentsIsNotEmpty(args);
+            return new AddCommand(parseEventArguments(args));
+        case "deadline":
+            verifyArgumentsIsNotEmpty(args);
+            return new AddCommand(parseDeadlineArguments(args));
+        case "delete":
+            verifyArgumentsIsNotEmpty(args);
+            return new DeleteCommand(parseTaskIndex(args));
+        case "find":
+            verifyArgumentsIsNotEmpty(args);
+            return new FindCommand(args);
+        case "":
+            return new EmptyCommand();
+        default:
+            return new UnknownCommand();
+        }
```

Comment:
Hi Wei Quan, I don't think this is a coding standard violation but I was wondering if it is better to declare constants for your respective cases? I suggest this because if you're managing a super large codebase and you somehow have to check for these `inputCommands` , a constant may save you the time taken to find these hard-coded commands. 

Let me know your thoughts! :)